### PR TITLE
Create a flag to filter out non-RPC networks for new and restored users.

### LINKED ISSUE #3910: Remove non-upstream networks: "Mainnet", "Rinkeby" and "Ropsten" from the network list
### Description

*Type*: Bug

*Summary*:  For Beta scope: remove options "Mainnet", "Rinkeby" and "Ropsten" from the network list. It does not do anything (no syncing too , so only networks with upstream RPC make sense now.

#### Expected behavior
Networks list in user Profile (Advanced) contains no options "Mainnet", "Rinkeby" and "Ropsten"
Default selected network: Mainnet with upstream RPC

### Reproduction

- Open Status
- Open Profile
- Tap Advanced
- Check if there is option "Mainnet" (or "Rinkeby" or  "Ropsten") in the networks list

### Additional Information
* Status version: nightly build Apr 17
* Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by mandrigin**
Upgrade users will have them in "Custom networks" section. 
Why isn't that a problem?

We force users to remove their app DB when upgrading to `0.9.19` due to wallet compatibility anyway, so everyone on alpha will be a new/restoring user.

fixes #3910 

status: ready

### CODE REVIEW

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/PRBf6B for branch PR-4390

<blockquote><div><strong><a href="https://i.diawi.com/PRBf6B">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/oGonpQ for branch PR-4390

<blockquote><div><strong><a href="https://i.diawi.com/oGonpQ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Review by yenda — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/M2Tgdt for branch PR-4390

<blockquote><div><strong><a href="https://i.diawi.com/M2Tgdt">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/constants.cljs [hunk 1]
```diff
@@ -68,9 +68,18 @@
...
+
+(defn filter-dict [func source-dict]
+  (reduce conj
+          {}
+          (filter func source-dict)))
+
 (def default-networks
-  (merge testnet-networks
-         mainnet-networks))
+  (filter-dict network-enabled? (merge testnet-networks mainnet-networks)))
```
Comment (flexsurfer) [inline src/status_im/constants.cljs]:
```clojure
(into {} (filter network-enabled? (merge testnet-networks mainnet-networks)))
```

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/Yztd91 for branch PR-4390

<blockquote><div><strong><a href="https://i.diawi.com/Yztd91">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/UQMLY7 for branch PR-4390

<blockquote><div><strong><a href="https://i.diawi.com/UQMLY7">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: src/status_im/constants.cljs (hunk 1 above)
Comment (mandrigin) [reply src/status_im/constants.cljs]: fixed

**Review by flexsurfer — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/n6rpE7 for branch PR-4390

<blockquote><div><strong><a href="https://i.diawi.com/n6rpE7">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/qtRLfp for branch PR-4390

<blockquote><div><strong><a href="https://i.diawi.com/qtRLfp">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

File: .env [hunk 2]
```diff
@@ -3,6 +3,7 @@ STUB_STATUS_GO=0
 ETHEREUM_DEV_CLUSTER=1
 MAINNET_WARNING_ENABLED=0
 OFFLINE_INBOX_ENABLED=1
+RPC_NETWORKS_ONLY 1
```
Comment (pedropombeiro) [inline .env:6]: @mandrigin legit question, any reason this line doesn't use the `=` sign like the rest?

Comment (dmitryn) [reply .env:6]: Seems like typo. When `RPC_NETWORKS_ONLY ` setting is not in `env` file, default value is still 1, that's why it's working even with this typo. /cc @mandrigin

Comment (pedropombeiro) [reply .env:6]: :+1: I'm working on a PR for https://github.com/status-im/status-react/issues/4396, I can include the fix there.

**Pull request merged**